internal/server/proto/server.proto

```diff
@@ -287,6 +287,53 @@ message Application {
   string file_change_signal = 3;
 }
 
+// Variable stores a variable's value, and metadata to allow for precedence 
+// sorting and source-specific error messaging 
+message Variable {
+  string name = 1;
+
+  oneof value {
+    string str = 2;
+    string hcl = 3;
+  }
+
+  // Source tracks where a value came from for more helpful error messaging.
+  // This is optional. If unset, we will give a generic error message.
+  oneof source {
+    SourceCLIArg cliarg = 4;
+    SourceCLIFile file = 5;
+    SourceEnvVar envvar= 6;
+    SourceRepoFile repofile = 7;
+    SourceServer server = 8;
+  }
+
+  // SourceCLIArg is set if a variable value is set via a -var flag
+  message SourceCLIArg {}
```

Comment:
You may be planning on adding fields later, but you can just use `google.protobuf.Empty` instead of creating an empty message. if we ever introduce structure later, this is completely backwards compatible since protobuf if structural/tag-based and doesn't care about the name of a thing.

Reply:
Nice, thanks!